> No if you read it carefully, it's point towards code that you wrote and explains how to fix it.

Okay, I see. It says:
```
************* File esphome/core/util.cpp
Error: ERROR esphome/core/util.cpp:71:2 - #define macros for integer constants are not allowed, please use static const uint8_t WEBSERVER_PORT = 80; style instead (replace uint8_t with the appropriate datatype). See also Google style guide.

Error: Process completed with exit code 1.
```

You are right, the define is my code:
```
+#ifndef WEBSERVER_PORT
+#define WEBSERVER_PORT 80
+#endif
```

I went for a define for a reason. The reason is that even without a configured webserver the mDNS server wants to make sure the the device is found. So it *always* announces a webserver service.
Therefore we have WEBSERVER_PORT defined to 80 as fallback.
The linter suggests to use a global static variable instead of the define. I don't see how that is better nor it should work.
Because this code *will* generate a define if the webserver component is configured, the linter just does not see it.
`cg.add_define('WEBSERVER_PORT', config[CONF_PORT])`

Maybe something like `cg.add_global()` would work, but only a variant that allows overriding an existing variable.

--- a/esphome/components/prometheus/__init__.py
+++ b/esphome/components/prometheus/__init__.py
@@ -18,5 +18,7 @@ CONFIG_SCHEMA = cv.Schema({
 def to_code(config):
     paren = yield cg.get_variable(config[CONF_WEB_SERVER_BASE_ID])
 
+    cg.add_define('USE_PROMETHEUS')
+
     var = cg.new_Pvariable(config[CONF_ID], paren)
     yield cg.register_component(var, config)

--- a/esphome/components/web_server/__init__.py
+++ b/esphome/components/web_server/__init__.py
@@ -36,6 +36,7 @@ def to_code(config):
     yield cg.register_component(var, config)
 
     cg.add(paren.set_port(config[CONF_PORT]))
+    cg.add_define('WEBSERVER_PORT', config[CONF_PORT])
     cg.add(var.set_css_url(config[CONF_CSS_URL]))
     cg.add(var.set_js_url(config[CONF_JS_URL]))
     if CONF_AUTH in config:

--- a/esphome/core/util.cpp
+++ b/esphome/core/util.cpp
@@ -43,6 +43,10 @@ bool network_is_connected() {
 bool mdns_setup;
 #endif
 
+#ifndef WEBSERVER_PORT
+static const uint8_t WEBSERVER_PORT = 80;
+#endif
+
 #ifdef ARDUINO_ARCH_ESP8266
 void network_setup_mdns(IPAddress address, int interface) {
   // Latest arduino framework breaks mDNS for AP interface
@@ -65,9 +69,9 @@ void network_setup_mdns(IPAddress address, int interface) {
       MDNS.addServiceTxt("esphomelib", "tcp", "mac", get_mac_address().c_str());
     } else {
 #endif
-      // Publish "http" service if not using native API.
+      // Publish "http" service if not using native API nor the webserver component
       // This is just to have *some* mDNS service so that .local resolution works
-      MDNS.addService("http", "tcp", 80);
+      MDNS.addService("http", "tcp", WEBSERVER_PORT);
       MDNS.addServiceTxt("http", "tcp", "version", ESPHOME_VERSION);
 #ifdef USE_API
     }

--- a/esphome/core/util.cpp
+++ b/esphome/core/util.cpp
@@ -76,6 +76,9 @@ void network_setup_mdns(IPAddress address, int interface) {
 #ifdef USE_API
     }
 #endif
+#ifdef USE_PROMETHEUS
+    MDNS.addService("prometheus-http", "tcp", WEBSERVER_PORT);
+#endif
   }
   void network_tick_mdns() {
 #ifdef ARDUINO_ARCH_ESP8266

